src/Common/Crypto/CryptoGen.php

```diff
@@ -267,16 +268,49 @@ private function coreDecrypt($sValue, $customPassword = null, $keySource = 'driv
                 $iv
             );
         } else {
-            error_log("OpenEMR Error : Decryption failed authentication.");
+            try {
+                // throw an exception
+                throw new Exception("OpenEMR Error: Decryption failed authentication!");
+            } catch (Exception $e) {
+                // log the exception message and call stack then return legacy null as false for
+                // those evaluating the return value as $return == false which with legacy will eval as false.
+                // I've seen this in the codebase, and it's a bit of a hack, but it's a way to return false instead of null.
+                // Dev's should use empty() instead of == false to check return from this function.
+                // The goal here is so the call stack is exposed to track back to where the call originated.
+                $stackTrace = debug_backtrace();
+                $formattedStackTrace = $this->formatExceptionMessage($stackTrace);
+                error_log(errorLogEscape($e->getMessage()) . "\n" . text($formattedStackTrace));
+                return false;
```

Comment:
Changing the behavior of this function to return false instead of null (ie. no explicit return) here will possibly break things and would likely require scanning all code calling the decryptStandard() function. Is this an important change?

Reply:
yes and I insist. It is already broken in API's because this the legacy return null value is being evaluated as returning false.
== null vs == false eval differently. I know to check around but either way it's already broken. I comment in code about the reasons for this.

Reply:
Okay, Thanks sir and I'm still testing so we'll see.